openjdk/src/main/java/org/conscrypt/Platform.java

```diff
@@ -543,9 +543,16 @@ public static String getOriginalHostNameFromInetAddress(InetAddress addr) {
             return originalHostName;
         } catch (InvocationTargetException e) {
             throw new RuntimeException("Failed to get originalHostName", e);
-        } catch (ReflectiveOperationException ignore) {
-            // passthrough and return addr.getHostAddress()
+        } catch (RuntimeException e) {
+            throw e;
+        } catch (Exception ignore) {
```

Comment:
This seems pretty unsafe. It looks like all the relevant exceptions exist in Java 6, but the convenience base class ReflectiveOperationException is new, is that right? In that case, it seems better to list out the exceptions and avoid the convenience base class.

Reply:
I believe this change is safe. The method was never declared to throw, which means any checked exceptions must have been caught previously.  `RuntimeException` is the base for unchecked exceptions, so I test that first.  Otherwise, everything else is a pass-through.

Reply:
Right, but now if the code changes later, we won't notice that we've missed an exception. But none of those methods are declared to throw ReflectiveOperationException, otherwise Java 6 code wouldn't compile under Java 7. Class.forName throws ClassNotFoundException, etc.

If you just list out all the exceptions, it's more code (the cost of this Java 6 silliness), but just as safe as it was before.

Reply:
@davidben it depends on what we're trying to achieve. I believe the goal here is to simply ignore any checked exception and drop down to the default approach (`RuntimeException`s are different, because they imply something more severe). If that's the case, then this is doing exactly that.

@flooey thoughts?